lib/msf/core/db_manager/session.rb

```diff
@@ -170,54 +115,129 @@ def report_session(opts)
   # @param wspace [Mdm::Workspace]
   # @return [void]
   def infer_vuln_from_session(session, wspace)
-    s = session.db_record
-    host = session.db_record.host
+    ::ActiveRecord::Base.connection_pool.with_connection {
+      s = session.db_record
+      host = session.db_record.host
+
+      if session.via_exploit == "exploit/multi/handler" and session.exploit_datastore['ParentModule']
+        mod_fullname = session.exploit_datastore['ParentModule']
+      else
+        mod_fullname = session.via_exploit
+      end
+      mod_detail = ::Mdm::Module::Detail.find_by_fullname(mod_fullname)
+      mod_name = mod_detail.name
+
+      vuln_info = {
+        exploited_at: Time.now.utc,
+        host: host,
+        info: "Exploited by #{mod_fullname} to create Session #{s.id}",
+        name: mod_name,
+        refs: mod_detail.refs.map { |ref| ref.name },
```

Comment:
This could be changed to `mod_detail.refs.map(&:name)` to be more compact and it would be faster too.